packages/@glimmer/compiler/lib/template-compiler.ts

```diff
@@ -127,24 +127,25 @@ export default class TemplateCompiler {
         this.attribute([typeAttr], hasSplat || actionIsComponent);
       }
 
+      for (let i = 0; i < action.modifiers.length; i++) {
+        this.modifier([action.modifiers[i]]);
+      }
+
       this.opcode(['flushElement', action], null);
     }
   }
 
   closeElement([action]: [AST.ElementNode]) {
-    if (isDynamicComponent(action)) {
-      this.opcode(['closeDynamicComponent', action], action);
-    } else if (isNamedBlock(action)) {
+    if (isNamedBlock(action)) {
       this.opcode(['closeNamedBlock', action]);
-    } else if (isComponent(action)) {
-      this.opcode(['closeComponent', action], action);
-    } else if (action.modifiers.length > 0) {
-      for (let i = 0; i < action.modifiers.length; i++) {
-        this.modifier([action.modifiers[i]]);
-      }
-      this.opcode(['closeElement', action], action);
     } else {
-      this.opcode(['closeElement', action], action);
+      if (isDynamicComponent(action)) {
```

Comment:
This can be restored to the `} else if(...) {` form, now that we aren't sharing any code between these branches

Reply:
Done